lib/core/src/lib/form/components/widgets/core/form-field.model.ts

```diff
@@ -513,6 +531,25 @@ export class FormFieldModel extends FormWidgetModel {
         return this.options?.length > 0;
     }
 
+    isEmptyValueOption(option: FormFieldOption): boolean {
+        return this.hasEmptyValue && option?.id === this.defaultEmptyOptionId;
+    }
+
+    private addOptions(options: FormFieldOption[]) {
+        options.forEach((option) => this.addOption(option));
+    }
+
+    private addOption(option: FormFieldOption) {
+        const alreadyExists = this.options.find((opt) => opt.id === option.id);
```

Comment:
in some methods where we operate on options we add optional chaining operator (not sure its needed, options might be falsy?), so to be consistent we should also add it here